sdv/single_table/base.py

```diff
@@ -305,6 +305,7 @@ def auto_assign_transformers(self, data):
             data (pandas.DataFrame):
                 The raw data (before any transformations) that will be used to fit the model.
         """
+        self.metadata._validate_metadata_matches_data(data)
```

Comment:
Should we use `validate()` here? I think it's the method called by `fit()`:
https://github.com/sdv-dev/SDV/blob/9421b160b8a2bd04950c305eee7244bc752b6806/sdv/single_table/base.py#L158